File: ios/ClaudeCursor/ClaudeCursor/Models/Session.swift
import Foundation

struct Session: Codable, Identifiable, Hashable {
    let name: String
    let status: String
    let shell: String?
    let port: Int?
    let pid: Int?
    let createdAt: String?
    let needsAction: Bool?
    let actionType: String?
    let needsActionSnippet: String?
    let confidence: Double?
    let description: String?

    var id: String { name }

    var isRunning: Bool { status == "running" }

    var actionRank: Int {
        guard needsAction == true else { return 10 }
        switch actionType {
        case "approval": return 0
        case "completed": return 1
        case "prompt": return 2
        case "flagged": return 3
        default: return 4
        }
    }
}

struct SessionsResponse: Codable {
    let sessions: [Session]
}

struct ShareInfo: Codable {
    let sessionName: String
    let status: String
}

File: android/ClaudeCursor/app/src/main/java/com/claudecursor/app/domain/model/Session.kt
package com.claudecursor.app.domain.model

data class Session(
    val name: String,
    val status: String,
    val shell: String? = null,
    val port: Int? = null,
    val pid: Int? = null,
    val createdAt: String? = null,
    val needsAction: Boolean = false,
    val actionType: String? = null,
    val needsActionSnippet: String? = null,
    val confidence: Double? = null,
    val description: String? = null
) {
    val isRunning: Boolean get() = status == "running"

    val actionRank: Int get() = if (!needsAction) 10 else when (actionType) {
        "approval" -> 0
        "completed" -> 1
        "prompt" -> 2
        "flagged" -> 3
        else -> 4
    }
}
